pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -465,9 +465,9 @@ private void downloadSegmentFromDeepStore(String segmentName, IndexLoadingConfig
    */
   private void untarAndMoveSegment(String segmentName, IndexLoadingConfig indexLoadingConfig, File segmentTarFile)
       throws IOException {
-    // TODO: This could leave temporary directories in _indexDir if JVM shuts down before the temporary directory is
-    //       deleted. Consider cleaning up all temporary directories when starting the server.
-    File tempSegmentDir = new File(_indexDir, "tmp-" + segmentName + "." + System.currentTimeMillis());
+    // This could leave temporary directories in _indexDir if JVM shuts down before the temp directory is deleted.
```

Comment:
If we really see a JVM gets shut down, I think it'd be good to consider changing JVM configs.

Reply:
What exactly do you have in mind here? I'm not sure what jvm config would make these code paths keep going and eventually delete